randovania/gui/multiplayer_session_window.py

```diff
@@ -887,6 +896,50 @@ async def export_game_button_clicked(self):
     async def _on_everyone_can_claim_check(self):
         await self.game_session_api.set_everyone_can_claim(self.everyone_can_claim_check.isChecked())
 
+    @asyncSlot()
+    async def _on_allow_coop_check(self):
+        if self.allow_coop_check.isChecked():
+            await async_dialog.message_box(
+                self,
+                QtWidgets.QMessageBox.Icon.Information,
+                "Information",
+                (
+                    "Co-op is still *very* experimental and may have issues.\nFor Prime 1 and Echoes in particular, "
+                    "please ensure that Randovania is always connected to the game before you collect items, as "
+                    "otherwise they will be lost permanently!"
+                ),
+            )
+        else:
+            world_count = collections.defaultdict(int)
```

Comment:
Maybe first iterate to create a `world_users: dict[WorldId, list[UserId]]`, so the unclaim loop at the end is simpler and `if is_world_assigned_to_multiple` could just be `if any(len(users) > 1 for users in world_users.values):`?

Reply:
yeah didn't occur to me. is indeed cleaner.